app/db/interface.py

```diff
@@ -46,8 +57,31 @@ def update_payment(self, payment_id, payment_replacement=None, payment_attribute
         :param payment_replacement: <dict> a complete payload that describes a new payload which replaces the old one
         :param payment_attributes: <dict> a collection of new payment attribute values that will overwrite old ones
         """
-
-        raise NotImplementedError()
+        
+        payment = Payment.query.get_or_404(payment_id)
+        if payment_replacement:
+            #need to handle is valid and throw 400 bad request for payment_replacement
+            payment.deserialize(payment_replacement)
+            self.db.session.commit()
```

Comment:
`@app.route('/pets/<int:id>', methods=['PUT'])
def update_pets(id):
    pet = Pet.query.get_or_404(id)
    pet.deserialize(request.get_json())
    db.session.commit()
    return make_response(jsonify(pet.serialize()), status.HTTP_200_OK)`
Referring to this piece of code from Professor's example, I don't think we would need to do an add since it's an update?